common/app/common/InlineStyles.scala

```diff
@@ -93,21 +94,48 @@ object InlineStyles {
   }
 
   /**
+    * Execute some code n times, or until no exception is thrown
+    * (If n is 0, code still gets executed once)
+    */
+  def retry[T](n: Int)(r: => T)(implicit onFail: (Throwable, Int) => Unit) = {
```

Comment:
The `dispatch` module, already included in this project, has a Retry mechanism. I don't think you need to rewrite your own.

It is used here for instance https://github.com/guardian/frontend/blob/master/onward/app/weather/WeatherApi.scala#L58